components/automate-ui/src/app/components/sidebar/sidebar.component.spec.ts

```diff
@@ -5,37 +5,79 @@ import { NgrxStateAtom, ngrxReducers, defaultInitialState, runtimeChecks } from
 
 import { SidebarComponent } from './sidebar.component';
 import { FeatureFlagsService } from 'app/services/feature-flags/feature-flags.service';
+import { LayoutFacadeService, Sidebar } from 'app/entities/layout/layout.facade';
+import { SettingsLandingComponent } from 'app/pages/settings-landing/settings-landing.component';
+import { using } from 'app/testing/spec-helpers';
+import { ComplianceLandingComponent } from 'app/pages/compliance-landing/compliance-landing.component';
 
 describe('SidebarComponent', () => {
   let store: Store<NgrxStateAtom>;
   let component: SidebarComponent;
   let fixture: ComponentFixture<SidebarComponent>;
+  let element: HTMLElement;
+  let layoutFacade: LayoutFacadeService;
+  let featureFlags: FeatureFlagsService;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ SidebarComponent ],
+      declarations: [SidebarComponent],
       schemas: [
         NO_ERRORS_SCHEMA,
         CUSTOM_ELEMENTS_SCHEMA
       ],
       providers: [
-        FeatureFlagsService
+        FeatureFlagsService,
+        LayoutFacadeService
       ],
       imports: [
         StoreModule.forRoot(ngrxReducers, { initialState: defaultInitialState, runtimeChecks })
       ]
     }).compileComponents();
     store = TestBed.inject(Store);
+    layoutFacade = TestBed.inject(LayoutFacadeService);
+    featureFlags = TestBed.inject(FeatureFlagsService);
     spyOn(store, 'dispatch').and.callThrough();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SidebarComponent);
     component = fixture.componentInstance;
+    element = fixture.debugElement.nativeElement;
+
+    // enable all feature flags, if any, for testing
+    featureFlags.setFeature('servicenow_cmdb', true);
+
     fixture.detectChanges();
   });
 
   it('should be created', () => {
     expect(component).toBeTruthy();
   });
+
+  using([
+    [new SettingsLandingComponent(), Sidebar.Settings, 'Settings Landing Component'],
+    [new ComplianceLandingComponent(), Sidebar.Compliance, 'Compliance Landing Component']
+
+  ], function (landingComponent: any, sidebar: Sidebar,  description: string) {
```

Comment:
Never, never introduce `any`! You will see me remark upon that in any code review I find it! 
Landing components have a base class (`LandingComponent`) but it is a _template_ base class, not a _code_ base class, so cannot do better than here any at present, sadly.